sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -482,6 +488,16 @@ pi_result piPlatformsGet(pi_uint32 NumEntries, pi_platform *Platforms,
     assert(ZeDriverCount == 1);
     ZE_CALL(zeDriverGet(&ZeDriverCount, &ZeDriver));
 
+    for (uint32_t i = 0; i < piPlatformsCache.size(); i++) {
+      if (piPlatformsCache[i]->ZeDriver == ZeDriver) {
+        Platforms[0] = piPlatformsCache[i];
+        if (NumPlatforms)
+          *NumPlatforms = 1;
```

Comment:
Curious about this code... Else?

Reply:
It is caller's responsibility to send a valid pointer value in NumPlatforms if they need to get this value set.
So, "else" means the caller did not want this info set, so skip setting this value.

Reply:
If you are talking about "if" at line #492, this means the cache did not find the match.
So, it will fall back to the original implementation below this code segment, which creates a new pi_platform and keep a copy in the cache.

Reply:
OK, it looks like I missed the goal of the un-commented API .
As usual feel free to add a lot of comments... :-)